ui/src/kapacitor/containers/KapacitorPage.js

```diff
@@ -71,6 +71,23 @@ class KapacitorPage extends Component {
     const {addFlashMessage, source, params, router} = this.props
     const {kapacitor} = this.state
 
+    let doesKapacitorNameAlreadyExist = false
+
+    if (source && source.kapacitors) {
+      source.kapacitors.map(existingKapacitor => {
+        if (existingKapacitor.name === kapacitor.name) {
+          doesKapacitorNameAlreadyExist = true
+        }
+      })
+    }
+    if (doesKapacitorNameAlreadyExist === true) {
+      addFlashMessage({
+        type: 'error',
+        text: `There is already a Kapacitor configuration named "${kapacitor.name}"`,
+      })
+      return
+    }
+
```

Comment:
Whenever I see a `initialize var`, `map over array to set var`, `switch on final var value`, I always try to rewrite it using a smarter enumerator. In this case, I think we can use `Array.prototype.some`:

```js
let nameExists
if (source && source.kapacitors) {
  nameExists = source.kapacitors.some(k => k.name === kapacitor.name)
}
if (nameExists) {
  ...
```